ouroboros-consensus-cardano/tools/db-analyser/Block/Cardano.hs

```diff
@@ -243,9 +243,9 @@ mkCardanoProtocolInfo ::
   -> SL.AlonzoGenesis
   -> Nonce
   -> NP STA (CardanoShelleyEras StandardCrypto)
-  -> ProtocolInfo IO (CardanoBlock StandardCrypto)
+  -> ProtocolInfo (CardanoBlock StandardCrypto)
 mkCardanoProtocolInfo genesisByron signatureThreshold genesisShelley genesisAlonzo initialNonce hardForkTriggers =
-    protocolInfoCardano
+    fst $ protocolInfoCardano @_ @IO
```

Comment:
Here's an example of where you have to specify `m` for GHC.

This is only necessary because in this case the arguments to `protocolInfoCardano` have no credentials and hence do not themselves fix `m`, right? In general, if you pass in credentials, then GHC will know the `m`.

Reply:
Yes, that's right.